demos/segmentation_demo_async/main.cpp

```diff
@@ -0,0 +1,235 @@
+    std::cout << "    -nstreams                 " << num_streams_message << std::endl;
+    std::cout << "    -loop                     " << loop_message << std::endl;
+    std::cout << "    -no_show                  " << no_show_processed_video << std::endl;
+    std::cout << "    -u                        " << utilization_monitors_message << std::endl;
+}
+
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    // ---------------------------Parsing and validation of input args--------------------------------------
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    slog::info << "Parsing input parameters" << slog::endl;
+
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+
+    if (FLAGS_m.empty()) {
+        throw std::logic_error("Parameter -m is not set");
+    }
+
+    return true;
+}
+
+void paintInfo(cv::Mat& frame, const PipelineBase::PerformanceInfo& info) {
+    std::ostringstream out;
+
+    out.str("");
+    out << "FPS:" << std::fixed << std::setprecision(0) << std::setw(3) << info.movingAverageFPS
+        << " (" << std::setprecision(1) << info.FPS << ")";
+    putHighlightedText(frame, out.str(), cv::Point2f(10, 30), cv::FONT_HERSHEY_TRIPLEX, 0.75, cv::Scalar(10, 200, 10), 2);
+
+    out.str("");
+    out << "Avg Latency:" << std::fixed << std::setprecision(0) << std::setw(4) << info.movingAverageLatencyMs
+        << " (" << std::setprecision(1) << info.getTotalAverageLatencyMs() << ") ms";
+    putHighlightedText(frame, out.str(), cv::Point2f(10, 60), cv::FONT_HERSHEY_TRIPLEX, 0.75, cv::Scalar(200, 10, 10), 2);
+
+    out.str("");
+    out << "Pool: " << std::fixed << std::setprecision(1) <<
+        info.numRequestsInUse << "/" << FLAGS_nireq;
+    putHighlightedText(frame, out.str(), cv::Point2f(10, 90), cv::FONT_HERSHEY_TRIPLEX, 0.75, cv::Scalar(200, 10, 10), 2);
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        /** This demo covers certain topology and cannot be generalized for any object detection **/
+        slog::info << "InferenceEngine: " << InferenceEngine::GetInferenceEngineVersion() << slog::endl;
+
+        // ------------------------------ Parsing and validation of input args ---------------------------------
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+
+        //------------------------------- Preparing Input ------------------------------------------------------
+        slog::info << "Reading input" << slog::endl;
+        auto cap = openImagesCapture(FLAGS_i, FLAGS_loop);
+        cv::Mat curr_frame;
+
+        //------------------------------ Running Segmentation routines ----------------------------------------------
+        PipelineBase pipeline(std::make_unique<SegmentationModel>(FLAGS_m),ConfigFactory::GetUserConfig());
```

Comment:
`auto_resize` is lost

Reply:
auto_resize was never implemented in segmentation demo, this parameter was in help by mistake, I've removed it.

Reply:
The aim is to work out general approach. In general nets should use auto_resize. That means you need to add the key.

Reply:
Segmentation model just don't have this flag. Only detection model had it originally - I've followed the same way. Approach is general, but flexible. I can add auto_resize here, but for now it's minor task and will be implemented in separate PR